heat/core/operations.py

```diff
@@ -123,13 +123,52 @@ def allclose(x, y, rtol = 1e-05, atol = 1e-08, equal_nan = False):
 
     """
 
-    if not isinstance(x, tensor.tensor):
-        raise TypeError('Expected x to be a ht.tensor, but was {}'.format(type(x)))
+    if isinstance(x, tensor.tensor):
+        pass
+
+    elif np.isscalar(x):
+        try:
+            x = tensor.array([float(x)])
+        except (ValueError, TypeError,):
+            raise TypeError('Data type not supported, input was {}'.format(type(x)))
+
+    else:
+        raise TypeError('Only tensors and numeric scalars are supported, but input was {}'.format(type(x)))
+
+    if isinstance(y, tensor.tensor):
```

Comment:
can be removed

Reply:
Why? Should there not be a check whether inputs are tensors? Or should this be merged into command if not isinstance(x, tensor.tensor) --> if np. isscalar ... else ...

Reply:
I see your point. But this code seems awkward with the empty condition. How about,

if np.isscalar...

elif not isinstance(x, tensor.tensor):
    raise ...

This would reduce the amount of code and complexity and would get rid of the empty clause.

Reply:
Jap, I agree... was not the prettiest way ;)

Reply:
Okay sounds good.